ports/webassembly/proxy_c.c

```diff
@@ -228,7 +241,15 @@ void proxy_c_to_js_lookup_attr(uint32_t c_ref, const char *attr_in, uint32_t *ou
         qstr attr = qstr_from_str(attr_in);
         mp_obj_t member;
         if (mp_obj_is_dict_or_ordereddict(obj)) {
-            member = mp_obj_dict_get(obj, MP_OBJ_NEW_QSTR(attr));
+            // Lookup the requested attribute as a dict key (not a method),
```

Comment:
that `not as method` kinda bothers me ... `function some({ thing = () => {} } = {}) {}` is a valid *JS* expectation, if this is covered in here I am OK with that comment (if explained somewhere else), otherwise I wonder "*why*"

Reply:
I think it's just that the comment I wrote here is confusing.  I meant that here the code is looking up the attribute as a key within the dict (or ordereddict).  That's opposed to the standard case of looking up the attribute as the name of a member/function/method of the target object.

Eg looking up `attr` on `target` will have the following logic:
- If `target` is a `dict` or `ordereddict` instance then lookup `target[attr]`, and return the result.  If the lookup failed, return `js.undefined`.
- Otherwise, lookup `target.foo`, raising an exception if it can't be found.

I've updated the comments to try and make it clearer.

I also added a test for passing through a function in the dict.

Reply:
Awesome, :+1: for me!